pint/utils.py

```diff
@@ -439,6 +440,74 @@ def is_number(s):
         pass
     return False
 
+def make_toas(startMJD, endMJD, ntoas, model, freq=1400, obs='GBT'):
+    '''make evenly spaced toas without errors'''
+    def get_freq_array(bfv, ntoas):
+        freq = np.zeros(ntoas)
+        num_freqs = len(bfv)
+        for ii, fv in enumerate(bfv):
+            freq[ii::num_freqs] = fv
+        return freq
+    
+    times = np.linspace(np.longdouble(startMJD)*u.d, np.longdouble(endMJD)*u.d, ntoas) * u.day
```

Comment:
If passing linspace Quantities worked then multiplying the result by u.day wouldn't. Don't multiply by u.d in here. In fact, better to accept startMJD and endMJD as Time objects and extract their MJDs here. Optionally allow bare floating-point as MJDs also.

Reply:
TODO added